### PR TITLE
Hygiene fixes in CI checks

### PR DESCRIPTION
**Pull request by aggarw13**
Update the CI workflow to remove dependency on 3rdparty coverage checker action, and instead use the FreeRTOS/CI-CD-Github-Actions repository. Also, update the formatting, doxygen and link verifier checks to use the actions repository.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -116,31 +112,27 @@ jobs:
...
-            url-check:
-  url-check:
-    runs-on: ubuntu-latest
+      - name: Check formatting
+        uses: FreeRTOS/CI-CD-Github-Actions/formatting@main
+        with:
+          path: ./
+          exclude-dirs: .git
+  link-verifier:
+    runs-on: ubuntu-18.04
```
Comment (paulbartell) [inline .github/workflows/ci.yml:121]: Should be ubuntu-latest right?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -116,31 +112,27 @@ jobs:
...
+        uses: actions/setup-python@v2
         with:
-            path: ./kernel
-      - name: URL Checker
-        run: |
-            bash kernel/.github/actions/url_verifier.sh kernel
+          python-version: '3.8'
+      - name: Check Links
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```
Comment (paulbartell) [inline .github/workflows/ci.yml:130]: Does the URL checker need a GITHUB_TOKEN?

**Review by archigup — APPROVED** [review]

File: .github/workflows/ci.yml (hunk 2 above)
Comment (aggarw13) [reply .github/workflows/ci.yml:130]: The link verifier script uses the `gh` client for checking Issues and PRs. We don't necessarily need it. https://github.com/FreeRTOS/CI-CD-Github-Actions/tree/main/link-verifier#pre-requisites

Comment (paulbartell) [reply .github/workflows/ci.yml:130]: Nevermind. I understand the reason this is required..

**Review by paulbartell — APPROVED** [review]

**Pull request merged**